### PR TITLE
[BC] Add an interactive only mode for env.py

### PR DESCRIPTION
**Pull request by tvmarino**
Added interactive only mode for env.py which compiles only using iclang instead of both clang and iclang.

### CODE REVIEW

File: compiler_opt/rl/env.py [hunk 1]
```diff
@@ -311,9 +313,12 @@ def _get_clang_generator(
     module = yield
     with clang_session(
         clang_path, module, task_type, interactive=True) as iclang:
-      with clang_session(
-          clang_path, module, task_type, interactive=False) as clang:
-        yield iclang, clang
+      if interactive_only:
+        yield iclang, iclang
```
Comment (mtrofin) [inline compiler_opt/rl/env.py:317]: Could we return iclang here, i.e. not a tuple, and detect that on the receiving end? or (iclang, None)?

Comment (tvmarino) [reply compiler_opt/rl/env.py:317]: The relative reward is computed, that is, the compute_relative_rewards function of env.py always expects two scores. I found this to be the cleanest change. Returning only iclang and None will end up being identical but would require multiple changes and I didn't want to break anything.

Comment (mtrofin) [reply compiler_opt/rl/env.py:317]: OK - can you update the documentation string explaining this, too? (i.e. it's not quite "`interactive_only: If set to true only iclang is yielded`" - looking at the docstring, the return isn't quite well specified actually, what the generator generates is what's interesting - could you basically make the doc "right"?)

File: compiler_opt/rl/env_test.py [hunk 2]
```diff
@@ -215,6 +215,32 @@ def test_env(self, mock_popen):
...
+      self.assertEqual(step.step_type, env.StepType.FIRST)
+
+      for step_itr in range(_NUM_STEPS - 1):
+        del step_itr
+        step = test_env.step(np.array([1], dtype=np.int64))
+        self.assertEqual(step.step_type, env.StepType.MID)
+
+      step = test_env.step(np.array([1], dtype=np.int64))
+      self.assertEqual(step.step_type, env.StepType.LAST)
+      self.assertEqual(step.reward, {'default': 0.})
```
Comment (mtrofin) [inline compiler_opt/rl/env_test.py:242]: is the only way we know it's interactive by seeing the reward be 0? Can we test that the 2 iclangs are the same, too?

Comment (tvmarino) [reply compiler_opt/rl/env_test.py:242]: We can directly check if test_env._iclang and test_env._clang are the same object.

File: compiler_opt/rl/env.py [hunk 3]
```diff
@@ -293,17 +293,19 @@ def _get_clang_generator(
...
   Args:
     clang_path: Path to the clang binary to use within InteractiveClang.
     task_type: Type of the MLGO task to use.
-    interactive_only: If set to true only iclang is yielded
+    interactive_only: If set to true the returned tuple of generators is
+      iclang, iclang instead of iclang, clang
 
   Returns:
-    The generator for InteractiveClang objects.
+    A tuple of generators created with clang_session. First argument of
```
Comment (mtrofin) [inline compiler_opt/rl/env.py:305]: nit: I think the return is a generator of tuples.

**Review by mtrofin — APPROVED** [review]
LGTM, nit in the comment

**Pull request merged**